packages/x-date-pickers/src/AdapterLuxon/AdapterLuxon.ts

```diff
@@ -199,7 +199,7 @@ export class AdapterLuxon implements MuiPickersAdapter<DateTime, string> {
   };
 
   public toISO = (value: DateTime) => {
-    return value.toISO({ format: 'extended' })!;
+    return value.toUTC().toISO({ format: 'extended' })!;
```

Comment:
Not related to the timezone, but while testing I struggled with this diff : the other libs show the UTC iso and Luxon did not